psi4/src/psi4/cc/amps.cc

```diff
@@ -0,0 +1,133 @@
+/*
+ * @BEGIN LICENSE
+ *
+ * Psi4: an open-source quantum chemistry software package
+ *
+ * Copyright (c) 2007-2018 The Psi4 Developers.
+ *
+ * The copyrights for code used from other parties are included in
+ * the corresponding files.
+ *
+ * This file is part of Psi4.
+ *
+ * Psi4 is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * Psi4 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with Psi4; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * @END LICENSE
+ */
+
+#include "psi4/cc/ccwave.h"
+#include "psi4/libpsio/psio.h"
+#include "psi4/liboptions/liboptions.h"
+#include "psi4/libmints/dimension.h"
+#include "psi4/libmints/matrix.h"
+#include "psi4/libpsi4util/process.h"
+#include <memory>
+namespace psi {
+namespace ccenergy {
+std::map<std::string, SharedMatrix> CCEnergyWavefunction::get_amplitudes() {
+    // re-init dpd
+    int** cachelist;
```

Comment:
You do not appear to free cache list, is this managed elsewhere?

Reply:
Thank you that is an oversight.